src/granite/transactions.cr

```diff
@@ -1,4 +1,6 @@
 module Granite::Transactions
+  class TransactionError < Exception; end
```

Comment:
Do we really want to create a custom exception class for this? I doubt it, but I'd like to hear from @drujensen on this.

If so, it needs to be named differently. This file is called 'transactions' but it doesn't yet do anything with the SQL concept of a transaction. It's not really a problem because the filename isn't presented to an appdev.

This exception will be handled and used directly by an appdev, and the name is misleading.

Reply:
>> Do we really want to create a custom exception class for this? I doubt it, but I'd like to hear from @drujensen on this.

@robacarp yeah, I think we do.  We have `Granite::Error` class to handle errors so I think having our own exception makes sense.  I would rename this to `Granite::Exception`.  Not sure I would place it here in this file.  Maybe in its own file `/src/granite/exception.cr`?

Reply:
I went [looking for the way ActiveRecord does this](http://api.rubyonrails.org/classes/ActiveRecord/Base.html#class-ActiveRecord::Base-label-Exceptions) and here's what I found.

- ActiveRecordError - Generic error class and superclass of all other errors raised by Active Record.
- RecordInvalid - raised by ActiveRecord::Base#save! and ActiveRecord::Base.create! when the record is invalid.
- RecordNotFound - No record responded to the ActiveRecord::Base.find method. Either the row with the given ID doesn't exist or the row didn't meet the additional restrictions. 
- StatementInvalid - The database server rejected the SQL statement. The precise error is added in the message.

After thinking on it, as an application developer, when I write the save! method, I want to be able to capture the specific exception that is going to be raised by save! and not other random exceptions.

I retract my statement about removing this custom exception. Instead I think that it should be a `Granite::InvalidRecord` exception. It's more helpful and explicit, and allows use of the save! / create! methods without opening up to a category of bugs about catching the wrong exception.

Reply:
( Rails `destroy!` raises a  ActiveRecord::RecordNotDestroyed )